QUESTIONS.md

```diff
@@ -0,0 +1,297 @@
+|-------------------------|---------------------------------------------------------------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| Difference way to write | ``` function sum(a, b) {    return a + b  } ``` <br> <br> ``` const sum = function sum(a, b) {    return a + b  } ```     | ``` const sum = (a, b) => {   return a + b } ``` <br> <br> ```  const sum = (a, b) => a + b ```                                                                                                        |
+| Arguments object        | ``` function regularFun() {    return Math.max(...arg); } ```                                                             | ``` const arrowFun = () => {   // If we try to access the arg    we're gonna get an exception } ``` <br> <br> // Right way to do <br> <br> ``` const arrowFun = (...numbers) =>  Math.max(numbers) ``` |
+| Prototype object        | ``` function regularFun() {} ``` <br> <br> ``` console.log(regularFun.prototype) ``` <br> <br> ``` { constructor: f } ``` | ``` const arrowFun = () => {} ``` <br> <br> ``` console.log(arrowFun.prototype)  ``` <br> <br> ``` undefined ```                                                                                       |
+
+**What is a higher order functions?**
+Basically, is a function that takes a function as an argument, or returns a function. 
+
+*Example using higher order function:*
+
+```
+const radius = [1, 2, 3]
+
+const area = function(radius) {
+    return (Math.PI * radius * radius)
+}
+
+const diameter = function(radius) {
+    return (2 * radius)
+}
+
+const calculate = function(radius, expression) {
+    const result = []
+    
+    for (let i = 0; i < radius.length; i++) {
+        result.push(expression(radius[i]))
+    }
+    
+    return result;
+}
+
+console.log(calculate(radius, area))
+```
+**What is the difference between filter and find?**
+
+With filter we can return an array that contain the element that satisfies the condition, but find returns an element that satisfies the condition
+
+The main different is what it return each method
+
+*Example using 'filter':*
+
+```
+const array = [1, 2, 3, 4, 5]
+
+const pair = array.filter(element => element % 2 == 0)
+
+console.log(pair)
+```
+
+*Example using 'find':*
+
+```
+const people = [
+    {
+        id: 1,
+        name: "Pablito"
+    },
+    {
+        id: 2,
+        name: "Juanito"
+    },
+    {
+        id: 3,
+        name: "Fulanito"
+    }
+]
+
+const person = people.find(element => element.id === 2)
+
+console.log(person)
+```
+
```

Comment:
Also remember that filter returns an empty array when no elements match the condition, while find returns undefined.